### ISSUE TITLE
List glitch with too many choices

### ISSUE THREAD

**Issue by gio00** [magmax/python-inquirer#38]
Hi, 

I found that the List glitch refreshing all the items if there are many. It is also depends on the size of terminal window though. 

The following example can show the problem:

```
    choices = []
    for x in range(0, 50):
        choices.append( x )

    questions = [
        inquirer.List('number',
            message="Pick a number:",
            choices=choices
        ),
    ]

    answer = inquirer.prompt(questions)
```

I am on Mac OS X 10.12.6
Python 2.7.14
inquirer 2.2.0

**Comment by magmax** [comment]
You are right there is a bug here, but maybe a list is not the best option here. What about using a Text question with a proper validation function? Something like:
```
inquirer.Text('number', message="Pick a number:",
                validate=lambda x, _: x.is_digit() and 0 <= int(x) = 50,
                )
```

**Comment by giorgiopogliani** [comment]
Yes, that would be nice but that code was just an example I am actually listing search results from a REST API, so for now I "fix" the problem limiting the search results

**Comment by gio00** [comment]
Yes, that would be nice but that code was just an example. I am actually listing search results from a REST API, so for now I "fixed" the problem limiting the search results

**Comment by iddoeldor91** [comment]
Any update ? having the same issue with checkbox

`print(inquirer.prompt([inquirer.Checkbox('a', message="message..", choices=[i for i in range(50)])]))`

**Comment by tterb** [comment]
@gio00 Have you managed to find any solution other than limiting the result count?

**Comment by gio00** [comment]
@tterb No, never really tried. I think the problem lays within the update of the first line where the selected item it's displayed

**Comment by maxcoldrick** [comment]
Just as quick workaround, I went with a pagination approach of splitting my choices list into a multidimensional structure of chunks[][] that would fit in the terminal window, appending a Next/Previous Page entry and navigating through the chunks[][] if Next or Previous is returned.
It's not perfect, but it mostly addresses the problem for me.

`
import inquirer
from blessings import Terminal

def handle_large_data(numbers, listLength):
    p=0
    listoflists=chunks(numbers, listLength)
    for j in listoflists:
        j.append("NEXT PAGE...")
        j.append("PREVIOUS PAGE...")

    answer=select_an_option(listoflists[p])
    the_chosen_number=answer['number']

    while the_chosen_number is "NEXT PAGE..." or "PREVIOUS PAGE...":
        if the_chosen_number is "NEXT PAGE...":
            nextPage=p+1
            if nextPage >= len(listoflists):
                p=0
                print "Back to top"
                answer=select_an_option(listoflists[p])
                the_chosen_number=answer['number']
            else:
                p+=1
                answer=select_an_option(listoflists[p])
                the_chosen_number=answer['number']

        elif the_chosen_number is "PREVIOUS PAGE...":
            p-=1
            answer=select_an_option(listoflists[p])
            the_chosen_number=answer['number']

        else:
            return the_chosen_number
            break

def select_an_option(choices):
    questions = [
        inquirer.List('number',
            message="Pick a number:",
            carousel=True,
            choices=choices
        ),
    ]

    answer = inquirer.prompt(questions)
    return answer

def chunks(l, n):
    n = max(1, n)
    return [l[i:i+n] for i in range(0, len(l), n)]

if __name__ == "__main__":
    # Create a list containing numbers 1-50
    choices = []
    for x in range(0, 50):
        choices.append( x )

    # Get height of terminal in characters, subtract 4 lines for display
    maximum_possible_list_length=(Terminal().height - 5)

    # If the amount of caretakers exceeds the height of the terminal, handle it
    if len(choices) > maximum_possible_list_length:
        the_chosen_number=handle_large_data(choices, maximum_possible_list_length)
    else:
        the_chosen_number=select_an_option(choices)

    print the_chosen_number
`

**Comment by magmax** [comment]
try commit 204f566e08e3b09ab0eca2dd77535458b6bdab9d (or master). It is a first approach, despite it still does something bad the corner-cases.

**Comment by orzechow** [comment]
It would be great to have a matrix like selection page as in zsh or scroll through the list as in the [Inquirer.js emoji plugin](https://github.com/tannerntannern/inquirer-emoji).

zsh:
![zsh completion matrix](http://lhg.hk/2015/02/24/enable-zsh-completion-of-brew-under-mac-os-x/youtube_dl_completion.png)

Inquirer.js emoji plugin:
![Inquirer.js emoji plugin](https://github.com/tannerntannern/inquirer-emoji/raw/master/demo.gif)

**Comment by AuHau** [comment]
Should be solved with #72
